libhtmlw/HTMLparse.c

```diff
@@ -309,14 +341,18 @@ ExpandEscapes(esc, endp, termination)
 			}
 			tchar = *tptr;
 			*tptr = '\0';
-			val = (char)atoi((esc + 1));
+			ucs = atoi((esc + 1));
+			lng=ucs2utf8(ucs,val);
+			fprintf(stderr,"&#%ud character: %s\n",ucs,val);
```

Comment:
Please fix formatting (missing spaces).  Also: is this debug printout really needed all the time?  Seems like it could become very noisy.

Reply:
Well the formatting is still rather weird (spaces are not rationed; I don't understand why you write statements without any spaces around the operators), but it's good enough maybe.